### PR TITLE
Machine ID: v16 upgrade guide

### PR DESCRIPTION
**Pull request by strideynet**
Add notes on the potentially breaking changes includes in v16.

### CODE REVIEW

**Comment by strideynet** [comment]
> Can we put a note in the changelog as well that links to this guide?

https://github.com/gravitational/teleport/pull/41482/commits/2a0b91263b7ac62b3bd7610773104f2e1dd00749

File: docs/pages/machine-id/reference/v16-upgrade-guide.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+title: Machine ID v16 Upgrade Guide
+description: Upgrade instructions for Machine ID in Teleport 16.0
+---
+
+Teleport 16.0 introduces a number of changes to Machine ID, some of which
+may require action on your part.
+
+You may need to take action if:
+
+- You use Machine ID with OpenSSH or OpenSSH based clients (e.g. Ansible).
```
Comment (ptgott) [inline docs/pages/machine-id/reference/v16-upgrade-guide.mdx:11]:
```suggestion
Teleport 16.0 introduces a number of changes to Machine ID. These changes may require action on your part if you use Machine ID with OpenSSH or OpenSSH-based clients (e.g. Ansible).

This guide explains how to migrate your Teleport 16.0 cluster for continued Machine ID support.
```

Suggested to condense the existing paragraphs and add a quick description of the scope of the guide.

**Review by ptgott — COMMENTED** [review]
Are we planning to add more H2 sections? If not, we can probably find a way to organize this guide so we don't need to go down to the H4 level.

**Review by ptgott — APPROVED** [review]

**Review by tigrato — COMMENTED** [review]
Can you leave a section to release notes so we won't forget them?

**Comment by strideynet** [comment]
@tigrato this has been added to the CHANGELOG.md in the implementation PR - https://github.com/gravitational/teleport/commit/2a0b91263b7ac62b3bd7610773104f2e1dd00749

File: docs/pages/machine-id/reference/v16-upgrade-guide.mdx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+file. You must do this before 17.0, but we recommend doing this in 16.0 to
+benefit from the performance improvements.
+
+To update your modified `ssh_config`, we recommend running an instance of `tbot`
+as you would usually configure it. This will generate a new `ssh_config`.
+Carefully inspect the `ProxyCommand` directive, and copy the new command to
+your modified `ssh_config`.
+
+You can find a full list of the parameters available for the
+`tbot ssh-proxy-command` on the [CLI reference page](/TODO).
```
Comment (tigrato) [inline docs/pages/machine-id/reference/v16-upgrade-guide.mdx:65]: Will you address this todo later?

**Review by tigrato — APPROVED** [review]

File: docs/pages/machine-id/reference/v16-upgrade-guide.mdx (hunk 2 above)
Comment (strideynet) [reply docs/pages/machine-id/reference/v16-upgrade-guide.mdx]: [ffec06c](https://github.com/gravitational/teleport/pull/41463/commits/ffec06c1adc2540847dfd9953b471f29f29eb18c)

**Pull request merged**